### PR TITLE
Add a new "reduced" absorption MGXS type

### PR DESCRIPTION
**Pull request by paulromano**
This PR adds a new "reduced" absorption cross section that can be used in downstream deterministic codes. This arose from a [conversation on our discourse group](https://openmc.discourse.group/t/different-k-eff-in-results-and-calculation/2181). This new MGXS type is the same as "RABSXS" in Serpent (discussed in [this paper](https://www.researchgate.net/publication/235958340_Revised_Methods_for_Few-Group_Cross_Section_Generation_in_the_Serpent_Monte_Carlo_Code)).

In addition to adding the new `ReducedAbsorptionXS` class, I've simplified many of the MGXS docstrings by extracting the common portion (parameters) and having it be applied automatically through a class decorator.

Would appreciate a review from one of our MGXS experts (@nelsonag @mkreher13) :pray:

### CODE REVIEW

File: openmc/mgxs/mgxs.py [hunk 1]
```diff
@@ -3224,103 +3067,72 @@ class AbsorptionXS(MGXS):
...
+        self._rxn_type = 'absorption'
 
-        .. versionadded:: 0.13.1
+
+@add_params
+class ReducedAbsorptionXS(MGXS):
+    r"""A reduced absorpiton multi-group cross section.
+
+    The reduced absorption reaction rate is defined as the difference between
+    absorption and the produced of neutrons due to (n,xn) reactions.
```
Comment (mkreher13) [inline openmc/mgxs/mgxs.py:3084]: the production of neutrons

Comment (mkreher13) [inline openmc/mgxs/mgxs.py:3084]: typo: the production of neutrons.

File: openmc/mgxs/mgxs.py [hunk 2]
```diff
@@ -3224,103 +3067,72 @@ class AbsorptionXS(MGXS):
...
+                 by_nuclide=False, name='', num_polar=1, num_azimuthal=1):
+        super().__init__(domain, domain_type, energy_groups, by_nuclide, name,
+                         num_polar, num_azimuthal)
+        self._rxn_type = 'absorption'
 
-        .. versionadded:: 0.13.1
+
+@add_params
+class ReducedAbsorptionXS(MGXS):
+    r"""A reduced absorpiton multi-group cross section.
```
Comment (mkreher13) [inline openmc/mgxs/mgxs.py:3081]: typo: absorption

**Review by mkreher13 — COMMENTED** [review]
This is a great addition and very timely for me as I have been looking into this question myself. The code looks straight-forward, but I still need to run it and check that the numbers add up on my end. I should be able to do that in the next few days.

**Comment by mkreher13** [comment]
I ran the example I wanted to test, and it is behaving as I would expect. I can match the Monte Carlo k-eff value with nu-fission/reduced-absorption and it is the same answer as using nu-fission/(total - consistent-nu-scatter).

**Comment by paulromano** [comment]
Thanks for taking the time to review this @mkreher13, really appreciate it!

**Pull request merged**